lib/metasploit/framework/data_service/remote/http/response_data_helper.rb

```diff
@@ -99,15 +99,17 @@ def process_file(base64_file, save_path)
   # @return [ActiveRecord::Base] A klass object, which inherits from ActiveRecord::Base.
   def to_ar(klass, val, base_object = nil)
     return nil unless val
-    data = val.class == Hash ? val.dup : JSON.parse(val)
+    data = val.class == Hash ? val.dup : JSON.parse(val, symbolize_names: true)
```

Comment:
Any reason to use `val.class == Hash` here rather than `val.is_a?(Hash)`?

Reply:
This was a copy/paste from somewhere else, so I'm not sure the original intent. One guess would be that `is_a?` will return `true` if `val` is a class that inherits from `Hash`. `val.class == Hash` will only return true if `val` is actually a `Hash`.

I'm going to leave it as-is since it's minor and could be intentional.